components/match2/commons/match_group_input.lua

```diff
@@ -583,7 +654,161 @@ function MatchGroupInput.readMvp(match)
 	return {players = parsedPlayers, points = mvppoints}
 end
 
+---@param match table
+---@param opponentIndex integer
+---@param opponentInput table
+---@return {players: table[], substitutions: MatchGroupInputSubstituteInformation[]}
+function MatchGroupInput.extractManualPlayersInput(match, opponentIndex, opponentInput)
+	--players from manual input as `opponnetX_pY`
+	local players = Array.mapIndexes(function(playerIndex)
+		return Json.parseIfString(match['opponent' .. opponentIndex .. '_p' .. playerIndex])
+	end)
+	local manualInput = {players = players}
+
+	local substitutions, parseFailure = Json.parseStringified(opponentInput.substitutes)
+	if parseFailure then
+		substitutions = {}
+	end
+
+	---@param playerData table|string|nil
+	---@return standardPlayer?
+	local getStandardPlayer = function(playerData)
+		if not playerData then return end
+		playerData = type(playerData) == 'string' and {playerData} or playerData
+		local player = {
+			displayName = Logic.emptyOr(playerData.displayName, playerData.displayname, playerData[1] or playerData.name),
+			pageName = Logic.emptyOr(playerData.pageName, playerData.pagename, playerData.link),
+			flag = playerData.flag,
+			faction = playerData.faction or playerData.race,
+		}
+		if Logic.isEmpty(player.displayName) then return end
+		player = PlayerExt.populatePlayer(player)
+		return player
+	end
+
+	manualInput.substitutions = Array.map(substitutions, function(substitution)
+		if type(substitution) ~= 'table' or not substitution['in'] then return end
+
+		return {
+			substitute = getStandardPlayer(substitution['in']),
```

Comment:
```suggestion
			substitute = getStandardPlayer(substitution.in),
```

Reply:
lets vscode think weird stuff:
![Screenshot 2024-08-15 140250](https://github.com/user-attachments/assets/f632d05e-604c-44c0-aec4-eea4ce9fdd55)

Reply:
Ah right, `in` is a lua keyword